test/irmin/test_tree.ml

```diff
@@ -542,6 +543,54 @@ let test_fold_force _ () =
       contents
   in
 
+  (* Ensure that [fold ~force:`True ~cache:false] visits newly added values. *)
+  let* () =
+    let* () = clear_and_assert_lazy sample_tree in
+    Tree.add sample_tree [ "a"; "ad" ] "v-ad" >>= fun updated_tree ->
+    let visited = ref false in
+    let contents k _ () =
+      if equal_key k [ "a"; "ad" ] then visited := true;
```

Comment:
```suggestion
      if equal_key k [ "a"; "ad" ] then (assert (not !visited); visited := true);
```

I think we can probably fold the last test into this one by asserting that our flag is only set once here. (Not sure if it's trying to test any other properties?)